packages/commerce-sdk-react/src/provider.tsx

```diff
@@ -73,8 +79,48 @@ const CommerceApiProvider = (props: CommerceApiProviderProps): ReactElement => {
         shopperOrders: new ShopperOrders(config),
         shopperProducts: new ShopperProducts(config),
         shopperPromotions: new ShopperPromotions(config),
-        shopperSearch: new ShopperSearch(config)
-    }
+        shopperSearch: new ShopperSearch(config),
+    })
+
+    useEffect(() => {
+        const newApiClients = {
+            shopperBaskets: new ShopperBaskets(config),
+            shopperContexts: new ShopperContexts(config),
+            shopperCustomers: new ShopperCustomers(config),
+            shopperDiscoverySearch: new ShopperDiscoverySearch(config),
+            shopperGiftCertificates: new ShopperGiftCertificates(config),
+            shopperLogin: new ShopperLogin(config),
+            shopperOrders: new ShopperOrders(config),
+            shopperProducts: new ShopperProducts(config),
+            shopperPromotions: new ShopperPromotions(config),
+            shopperSearch: new ShopperSearch(config),
+        }
+        setApiClients(newApiClients)
+    }, [clientId, organizationId, shortCode, siteId, proxy])
+
+    const [auth, setAuth] = useState(
+        new Auth({
+            clientId,
+            organizationId,
+            shortCode,
+            siteId,
+            proxy,
+            redirectURI,
+        })
+    )
+
+    useEffect(() => {
+        const newAuth = new Auth({
+            clientId,
+            organizationId,
+            shortCode,
+            siteId,
+            proxy,
+            redirectURI,
+        })
+        newAuth.ready()
+        setAuth(newAuth)
+    }, [clientId, organizationId, shortCode, siteId, proxy, redirectURI])
```

Comment:
Looks like this `useState` and `useEffect` combo is there for re-initializing the auth object _only_ when certain dependencies change. 

How about we use `useMemo` instead? So that the initialization code is not duplicated. For example:

```suggestion
    const auth = useMemo(() => {
        const newAuth = new Auth({
            clientId,
            organizationId,
            shortCode,
            siteId,
            proxy,
            redirectURI
        })
        newAuth.ready()

        return newAuth
    }, [clientId, organizationId, shortCode, siteId, proxy, redirectURI])
```